### DISCUSSION TITLE
How does BrewTestBot work and when does it update formulae?

### DISCUSSION THREAD

**Discussion by alexjpwalker** [Homebrew/discussions#3083] [Tap maintenance and Homebrew development] [answered]
I am one of the maintainers of the TypeDB project (homebrew formula `typedb`). We requested the addition of this formula in about 2018, but have not requested any updates since then.

Nonetheless, the file https://github.com/Homebrew/homebrew-core/blob/master/Formula/typedb.rb has been regularly updated with new releases from the `vaticle/typedb` GitHub repo. This is great!

Until about Jan 2022, most of those updates were authored by real people; since then, they have been authored and committed by https://github.com/BrewTestBot. How does BrewTestBot work? How does it determine when to upgrade packages? Is it simply listening for new GitHub releases?

**Comment by carlocab** [comment, marked as answer]
BrewTestBot runs `brew livecheck` on a schedule to check for outdated formula and then runs `brew bump-formula-pr` when it spots one that is outdated.

How `livecheck` works depends on the formula. For `typedb` it actually parses your repo's tags to check for new versions. This sometimes breaks when unconventional tags are pushed to a repository (e.g. a non-release tag of the date at the time of tagging), but we can apply formula-specific fixes when we encounter those.

**Reply by lolski** [reply]
I see! That's pretty cool.

How does BrewTestBot decide which packages it should check? Does it do it for every package in Homebrew Core?

**Reply by SMillerDev** [reply]
It's defined here: https://github.com/Homebrew/homebrew-core/blob/master/.github/workflows/autobump.yml based on packages that require few modifications on update

**Reply by lolski** [reply]
Got it. What are the criteria for a package to be added, and remain, on that list?

**Reply by SMillerDev** [reply]
Someone making a pull request to add or remove it and one of the Maintainers merging it. There are no defined criteria, but in an ideal world everything would be automatically updated of course.

**Reply by lolski** [reply]
Got it! I think we understand how everything works now. Thanks!
